codegen/src/lib.rs

```diff
@@ -55,47 +59,117 @@ pub fn deflate_file(ts: TokenStream) -> TokenStream {
 /// - The compile errors in `deflate_file!`
 /// - If the file contents are not all valid UTF-8
 #[proc_macro]
+#[proc_macro_error]
 pub fn deflate_utf8_file(ts: TokenStream) -> TokenStream {
     match inner(ts, true) {
         Ok(ts) => ts.into(),
         Err(err) => err.to_compile_error().into(),
     }
 }
 
-fn inner(ts: TokenStream, utf8: bool) -> Result<impl Into<TokenStream>> {
+/// An arguments expected provided by the proc-macro.
+///
+/// ```ignore
+/// flate!(pub static DATA: [u8] from "assets/009f.dat"); // default, DEFLATE
+/// flate!(pub static DATA: [u8] from "assets/009f.dat" with zstd); // Use Zstd for this file spcifically
+/// flate!(pub static DATA: [u8] from "assets/009f.dat" with deflate); // Explicitly use DEFLATE.
+/// ```
+struct FlateArgs {
+    path: syn::LitStr,
+    algorithm: Option<syn::Ident>,
+}
+
+impl syn::parse::Parse for FlateArgs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let path = input.parse()?;
+        let algorithm = input.parse().ok();
```

Comment:
Did you forget to parse `with` here?

Reply:
Nope, `with` is not passed to this proc-macro.